.github/workflows/sync-apis.yml

```diff
@@ -0,0 +1,51 @@
+# This workflow will review the current status of the APIs.
+# If there are updates in the APIs, it will re-generate the clients and open a PR with the change.
+
+name: Syncing APIs and Generating Clients
+
+on:
+  schedule:
+    - cron: "0 0 * * *"
+  push:
+    branches: [ "main" ]
+
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [18.x]
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        token: ${{ secrets.BOT_GH_TOKEN }}
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+    - name: Install Java and Maven
+      uses: actions/setup-java@4
+      with:
+        distribution: 'temurin'
+        java-version: '21'
+        cache: 'maven'
+    - run: npm install
+    - name: Generate clients
+      run: npm run nx:reset && npm run generate
+    - name: Build packages
+      run: npm run build:no-cache
+    - name: Create Pull Request
+      uses: peter-evans/create-pull-request@v4
+      with:
+        token: ${{ secrets.BOT_GH_TOKEN }}
+        title: Syncing APIs and Generating Clients
+        body: |
+          There were some changes in the contents of the APIs.
+
+          Automated changes by [create-pull-request](https://github.com/peter-evans/create-pull-request) GitHub action
+        commit-message: Syncing API
```

Comment:
@tahmidefaz FYI this commit message differs from what you expect above in the workflow

See https://github.com/RedHatInsights/javascript-clients/pull/300/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR72

We may also want to consider using convention commit syntax for this so we avoid someone making the same commit message on accident (something like chore: or release: ). See https://www.conventionalcommits.org/en/v1.0.0/#summary

Reply:
Done, replaced it with a `chore` prefix.